src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.Shared/Harness/MarshalledObjects.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for more information.
+
+namespace Xunit.Harness
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Runtime.Remoting;
+
+    internal sealed class MarshalledObjects : IDisposable
```

Comment:
I guess the only problem with this is that objects will live slightly longer than they need to (until `MarshalledObjects` is disposed instead of when they're done), but seems worth it to keep things clean.

Reply:
The life is otherwise controlled by a _timeout_, as opposed to GC reachability. This means long-running tests can cause harness objects to be cleaned up even though they are still going to be used.